Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -365,55 +368,172 @@ def start_chrome_headless(chrome_port, chrome_binary=CHROME_EXE, user_options=""
+
+    elif instance_id in instances_id and chrome_options == instance_id_to_chrome_options.get(instance_id):
+        """
+        When:
+            case 5: The instance exists in the file and the chrome configuration match to it
+        Then:
+            use the existing browser
+        """
+
+        demisto.debug(f"case 5: found {instance_id=} with matching {chrome_options=}")
+        chrome_port = chrome_options_to_port.get(chrome_options)
+        browser = is_chrome_running_locally(chrome_port)
+        return browser, chrome_port
 
+    else:
+        message = f"ERROR: Cannot connect to Chrome with parameters: {instance_id=}, {chrome_options=}"
+        demisto.info(message)
+        demisto.error(message)
+
+
+def get_chrome_instances_contents_dictionaries(chrome_instances_contents):
+    instance_id_to_chrome_options = {}
+    chrome_options_to_port = {}
+    instance_id_to_port = {}
+
+    if chrome_instances_contents:
+        splitted_chrome_instances_contents = chrome_instances_contents.strip().splitlines()
+        for line in splitted_chrome_instances_contents:
+            port, instance_id, chrome_options = line.strip().split('\t')
+            instance_id_to_chrome_options[instance_id] = chrome_options
+            chrome_options_to_port[chrome_options] = port
+            instance_id_to_port[instance_id] = port
+
+    return instance_id_to_chrome_options, chrome_options_to_port, instance_id_to_port
+
+
+def generate_new_chrome_instance(instance_id, chrome_options):
+    chrome_port = generate_chrome_port()
+    if chrome_port:
+        browser, chrome_port = start_chrome_headless(str(chrome_port), instance_id, chrome_options)
+        return browser, chrome_port
+    return None, None
+
+
+def generate_chrome_port():
     first_chrome_port = FIRST_CHROME_PORT
     ports_list = list(range(first_chrome_port, first_chrome_port + MAX_CHROMES_COUNT))
     random.shuffle(ports_list)
     demisto.debug(f"Searching for Chrome on these ports: {ports_list}")
-
     for chrome_port in ports_list:
         len_running_chromes = count_running_chromes(chrome_port)
-        demisto.debug(f"Found {len_running_chromes=} on port {chrome_port} has ")
+        demisto.debug(f"Found {len_running_chromes=} on port {chrome_port}")
 
         if len_running_chromes == 0:
             # There's no Chrome listening on that port, Start a new Chrome there
-            demisto.debug(f"No Chrome found on port {chrome_port}")
-            demisto.debug(f'Initializing a new Chrome session on port {chrome_port}')
-            browser, chrome_port = start_chrome_headless(str(chrome_port))
-            if browser:
-                return browser, chrome_port
-
-        # There's already a Chrome listening on that port, Don't use it
+            demisto.debug(f"No Chrome found on port {chrome_port}, using it.")
+            return chrome_port
 
+    # There's already a Chrome listening on that port, Don't use it
```

Comment:
That comment doesn't seem to belong in this place.
You're giving up on the next line.